### PR TITLE
Fixed Dimension Mismatch - AbtractMatrix and OneHotVector

### PR DESCRIPTION
**Pull request by maerory**
Fixed the multiplication function so that it only works when the column of matrix matches with the size of onehotvector.

### PR Checklist

- [ ] Tests are added
- [ ] Entry in NEWS.md
- [ ] Documentation, if applicable
- [ ] Final review from `@MikeInnes` or `@dhairyagandhi96` (for API changes).

### CODE REVIEW

File: src/onehot.jl [hunk 1]
```diff
@@ -11,7 +11,12 @@ Base.getindex(xs::OneHotVector, i::Integer) = i == xs.ix
 
 Base.getindex(xs::OneHotVector, ::Colon) = OneHotVector(xs.ix, xs.of)
 
-A::AbstractMatrix * b::OneHotVector = A[:, b.ix]
+function Base.:*(A::AbstractMatrix, b::OneHotVector)
+  if size(A)[2] != b.of
```
Comment (CarloLucibello) [inline src/onehot.jl:15]:
```suggestion
  if size(A, 2) != b.of
```

**Comment by CarloLucibello** [comment]
Can we add a `@test_throws DimensionMismatch ...`    test here?

**Comment by DhairyaLGandhi** [comment]
Generally sgtm

**Comment by maerory** [comment]
> Can we add a `@test_throws DimensionMismatch ...` test here?

Do you mean unit tests?

**Comment by CarloLucibello** [comment]
yes, you can add a test here https://github.com/FluxML/Flux.jl/blob/master/test/onehot.jl

**Comment by CarloLucibello** [comment]
@maerory  bump

**Comment by maerory** [comment]
Sorry about the delay,

I have added a new tester, please check whether it is enough or I should add more.
In addition, I have discovered another error regarding OneHotVector creation.

When OneHotVector(i,  n) is called where i > n, Flux is currently returning zero vector of size n.
Should this also throw an error?
I will open another issue and fix this if that is the case.

**Comment by maerory** [comment]
@CarloLucibello bump+

**Comment by CarloLucibello** [comment]
Thanks!

bors r+

**Pull request merged**